tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,274 @@
+from composer.models import ModelHparams
+from composer.optim import AdamHparams, ExponentialSchedulerHparams
+from composer.trainer.devices import CPUDeviceHparams, DeviceHparams, GPUDeviceHparams
+from composer.trainer.trainer_hparams import TrainerHparams, algorithms_registry
+from composer.utils import dist, is_tar
+from tests.trainer.test_checkpoint import assert_checkpoints_equivalent
+from tests.utils.synthetic_utils import configure_dataset_for_synthetic, configure_model_for_synthetic
+
+algo_hparams_overrides = {
+    "swa": {
+        "swa_start": "0.4dur",
+        "swa_end": "0.8dur"
+    },
+    "cutmix": {
+        "num_classes": None
+    },
+    "ghost_batchnorm": {
+        "ghost_batch_size": 2
+    },
+    "stochastic_depth": {
+        "target_layer_name": "ResNetBottleneck"
+    }
+}
+
+xfail_list = [
+    "blurpool",  # Does not fail w/ deepspeed
+    "layer_freezing",
+    "factorize",
+    "sam",
+    "squeeze_excite",
+    "stochastic_depth",  # timeout
+    "swa"
+]
+
+skiplist = {
+    'default': {
+        'colout': 'Only for images',
+        'cutmix': 'Only for images',
+        'cutout': 'Only for images',
+        'alibi': 'Not compatible with simple model.',
+        'seq_length_warmup': 'Not compatible with simple model.',
+        'randaugment': 'Requires PIL dataset to test.',
+        'augmix': 'Required PIL dataset to test.',
+        'stochastic_depth': 'Only applies to ResNets.',
+        'no_op_model': 'Not compatible with this model.'
+    },
+    'resnet50_synthetic': {
+        'alibi': 'Not compatible with vision model.',
+        'seq_length_warmup': 'Not compatible with vision model.',
+        'randaugment': 'Requires PIL dataset to test.',
+        'augmix': 'Required PIL dataset to test.',
+        'no_op_model': 'Not compatible with this model.'
+    },
+    'gpt2_52m': {
+        'blurpool': 'Only for CNNs',
+        'channels_last': 'Only for CNNs',
+        'colout': 'Only for images',
+        'cutmix': 'Only for images',
+        'cutout': 'Only for images',
+        'squeeze_excite': 'Not defined for language model transformers',
+        'mixup': 'Only defined for images',
+        'randaugment': 'Requires PIL dataset to test.',
+        'augmix': 'Required PIL dataset to test.',
+        'progressive_resizing': 'Only defined for images',
+        'stochastic_depth': 'Only applies to ResNets.',
+        'no_op_model': 'Not compatible with this model.'
+    }
+}
+
+
+@pytest.mark.timeout(180)
+@pytest.mark.parametrize("world_size", [
+    pytest.param(1),
+    pytest.param(2, marks=pytest.mark.world_size(2)),
+])
+@pytest.mark.parametrize("device_hparams,deepspeed_enabled,zero_stage", [
+    pytest.param(CPUDeviceHparams(), False, None, id="cpu-ddp"),
+    pytest.param(GPUDeviceHparams(), False, None, id="gpu-ddp", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 0, id="deepspeed-zero0", marks=pytest.mark.gpu),
```

Comment:
Is it necessary to check serialization on deepspeed? I don't think deepspeed would change anything w.r.t. an individual algorithm (the standard DDP tests ensure that algorithms do not assume that the model is a ComposerModel)